### PR TITLE
vlc: fix headphones mode; libspatialaudio: init at 0.3.0

### PR DESCRIPTION
**Pull request by krav**
###### Motivation for this change

The `binauralizer` aout filter is currently missing, causing menu option "Audio -> Stereo Mode -> Headphones" to log an error and fall back to original audio. It depends on libspatialaudio which is added in this commit, and a "Head-Related Transfer Function" file that's skipped by `make install`.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/libspatialaudio/default.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    hash = "sha256-sPnQPD41AceXM4uGqWXMYhuQv0TUkA6TZP8ChxUFIoI=";
+  };
+  buildInputs = [ cmake libmysofa zlib ];
+  configurePhase = "cmake -DCMAKE_INSTALL_PREFIX=$out .";
+
+  meta = {
+    description =
+      "Ambisonic encoding / decoding and binauralization library in C++";
+    homepage = "https://github.com/videolabs/libspatialaudio";
+    license = lib.licenses.lgpl21Plus;
```
Comment (thiagokokada) [inline pkgs/development/libraries/libspatialaudio/default.nix:17]: Can you add yourself as a maintainer :thinking: ?

File: pkgs/development/libraries/libspatialaudio/default.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{ lib, stdenv, fetchFromGitHub, cmake, libmysofa, zlib }:
```
Comment (thiagokokada) [inline pkgs/development/libraries/libspatialaudio/default.nix:1]: Sorry about this, but this looks really strange. Like a giraffe without neck or something.

```suggestion
{ lib, stdenv, fetchFromGitHub, cmake, libmysofa, zlib }:

```

File: pkgs/development/libraries/libspatialaudio/default.nix [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+stdenv.mkDerivation {
+  name = "libspatialaudio";
+  src = fetchFromGitHub {
+    owner = "videolabs";
+    repo = "libspatialaudio";
+    rev = "0.3.0";
+    hash = "sha256-sPnQPD41AceXM4uGqWXMYhuQv0TUkA6TZP8ChxUFIoI=";
+  };
+  buildInputs = [ cmake libmysofa zlib ];
+  configurePhase = "cmake -DCMAKE_INSTALL_PREFIX=$out .";
```
Comment (jtojnar) [inline pkgs/development/libraries/libspatialaudio/default.nix:11]: The `cmake` package contains a setup hook that should do something like this automatically.

File: pkgs/development/libraries/libspatialaudio/default.nix [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{ lib, stdenv, fetchFromGitHub, cmake, libmysofa, zlib }:
+stdenv.mkDerivation {
+  name = "libspatialaudio";
+  src = fetchFromGitHub {
+    owner = "videolabs";
+    repo = "libspatialaudio";
+    rev = "0.3.0";
+    hash = "sha256-sPnQPD41AceXM4uGqWXMYhuQv0TUkA6TZP8ChxUFIoI=";
+  };
+  buildInputs = [ cmake libmysofa zlib ];
```
Comment (jtojnar) [inline pkgs/development/libraries/libspatialaudio/default.nix:10]: Build tools like `cmake` should go to `nativeBuildInputs`, see https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md.

File: pkgs/development/libraries/libspatialaudio/default.nix [hunk 5]
```diff
@@ -0,0 +1,21 @@
+{ lib, stdenv, fetchFromGitHub, cmake, libmysofa, zlib }:
+stdenv.mkDerivation {
+  name = "libspatialaudio";
```
Comment (jtojnar) [inline pkgs/development/libraries/libspatialaudio/default.nix:3]: Could you use `pname` and `version`?

File: pkgs/development/libraries/libspatialaudio/default.nix (hunk 3 above)
Comment (krav) [reply pkgs/development/libraries/libspatialaudio/default.nix]: Indeed it does, not sure what I was thinking putting `cmake .` as `configurePhase` anyway. Removing this line makes it all good.

File: pkgs/development/libraries/libspatialaudio/default.nix (hunk 4 above)
Comment (krav) [reply pkgs/development/libraries/libspatialaudio/default.nix]: Fixed, thanks!

File: pkgs/development/libraries/libspatialaudio/default.nix (hunk 1 above)
Comment (krav) [reply pkgs/development/libraries/libspatialaudio/default.nix:22]: Ok :) I've subscribed to updates to their repo.

File: pkgs/development/libraries/libspatialaudio/default.nix (hunk 5 above)
Comment (krav) [reply pkgs/development/libraries/libspatialaudio/default.nix]: Yup!

File: pkgs/development/libraries/libspatialaudio/default.nix (hunk 2 above)
Comment (krav) [reply pkgs/development/libraries/libspatialaudio/default.nix]: Hah, someone needs to add this to nixfmt :P

File: pkgs/development/libraries/libspatialaudio/default.nix [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  meta = {
+    description =
+      "Ambisonic encoding / decoding and binauralization library in C++";
+    homepage = "https://github.com/videolabs/libspatialaudio";
+    license = lib.licenses.lgpl21Plus;
+    platforms = lib.platforms.linux;
+    maintainers = with lib.maintainers; [ krav ];
+  };
+}
+
```
Comment (oxalica) [inline pkgs/development/libraries/libspatialaudio/default.nix:27]:
```suggestion
```

**Review by oxalica — CHANGES_REQUESTED** [review]
There are some trivial suggestions.

I successfully built `vlc` but not sure how to test it. I only have these modes in `Audio -> Stereo Mode`.

Screenshot_20211217_232906

**Comment by krav** [comment]
> There are some trivial suggestions.

Thanks, I'll fix it.

> I successfully built `vlc` but not sure how to test it. I only have these modes in `Audio -> Stereo Mode`.

Seems that's what the menu looks like if you don't open material with surround sound. You should see all the available modes in Settings under `Audio -> Stereo audio output mode` after selecting showing `All` settings. [Here are some files to test with](https://www2.iis.fraunhofer.de/AAC/multichannel.html).

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 150864` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>libsForQt512.elisa</li>
    <li>libsForQt512.phonon-backend-vlc</li>
    <li>libsForQt514.elisa</li>
    <li>libsForQt514.phonon-backend-vlc</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>obs-studio-plugins.obs-ndi</li>
  </ul>
</details>
<details>
  <summary>28 packages built:</summary>
  <ul>
    <li>arcan.all-wrapped</li>
    <li>arcan.arcan</li>
    <li>arcan.arcan-wrapped</li>
    <li>arcan.durden-wrapped</li>
    <li>arcan.pipeworld-wrapped</li>
    <li>arcan.prio-wrapped</li>
    <li>arcan.xarcan</li>
    <li>kaffeine</li>
    <li>libsForQt5.elisa (libsForQt515.elisa ,plasma5Packages.elisa)</li>
    <li>libsForQt5.phonon-backend-vlc (libsForQt515.phonon-backend-vlc ,plasma5Packages.phonon-backend-vlc)</li>
    <li>libspatialaudio</li>
    <li>libvlc</li>
    <li>megaglest</li>
    <li>minitube</li>
    <li>obs-studio</li>
    <li>obs-studio-plugins.looking-glass-obs</li>
    <li>obs-studio-plugins.obs-gstreamer</li>
    <li>obs-studio-plugins.obs-move-transition</li>
    <li>obs-studio-plugins.obs-multi-rtmp</li>
    <li>obs-studio-plugins.obs-nvfbc</li>
    <li>obs-studio-plugins.obs-websocket</li>
    <li>obs-studio-plugins.wlrobs</li>
    <li>pympress</li>
    <li>python38Packages.python-vlc</li>
    <li>python39Packages.python-vlc</li>
    <li>reaper</li>
    <li>strawberry</li>
    <li>vlc (vlc_qt5)</li>
  </ul>
</details>

**Review by thiagokokada — APPROVED** [review]
> Seems that's what the menu looks like if you don't open material with surround sound. You should see all the available modes in Settings under Audio -> Stereo audio output mode after selecting showing All settings. Here are some files to test with.

Tested the headphone mode, seems to work fine (the effect is not that impressive but works).

**Pull request merged**